src/librbd/api/Mirror.cc

```diff
@@ -3150,21 +3128,22 @@ int Mirror<I>::group_promote(IoCtx& group_ioctx, const char *group_name,
     for (; snap != snaps.rend(); ++snap) {
       auto snap_type = cls::rbd::get_group_snap_namespace_type(
           snap->snapshot_namespace);
-      if (snap_type != cls::rbd::GROUP_SNAPSHOT_NAMESPACE_TYPE_MIRROR ||
-          snap->state == cls::rbd::GROUP_SNAPSHOT_STATE_INCOMPLETE) {
-        continue;
+      if (snap_type == cls::rbd::GROUP_SNAPSHOT_NAMESPACE_TYPE_MIRROR &&
+          snap->state == cls::rbd::GROUP_SNAPSHOT_STATE_COMPLETE) {
+        break;
       }
-      break;
     }
 
     if (snap == snaps.rend()) {
       lderr(cct) << "failed to rollback, no mirror group snapshot available"
-        << dendl;
+                 << dendl;
       return -EINVAL;
     }
+
     // Check for group membership match
     std::vector<cls::rbd::GroupImageSpec> rollback_images;
     for (auto& it : snap->snaps) {
+      rollback_snap_ids.push_back(it.snap_id);
       rollback_images.emplace_back(it.image_id, it.pool);
     }
 
```

Comment:
Add a log message here with the snap id of the group snap it will be rolled back to. Suggestion :
"rolling back to group snap id: "